### PR TITLE
Fix russian locale regexps

### PR DESCRIPTION
**Pull request by socketpair**
Many bugs including:

1. typos in regexp between `\.` and `.`
2. correct parsing of (for example on February)  `фев`, `февр`, `фев.` and `февр.`
3. fix order of moths in regexp, and eliminate duplicates

Since there is no docs that describes when `monthsShortStrictRegex` will be used, fix minor bugs. Still not sure if correct (if full)

### CODE REVIEW

**Comment by Oire** [comment]
@socketpair, what about the tests? Now they are failing.

**Comment by ichernev** [comment]
@socketpair the month regexes need to be sorted from increasing to decreasing length so it will match longest possible string.

**Comment by maggiepint** [comment]
Any progress on this? If I don't hear back soon will close due to the failing tests.

**Comment by socketpair** [comment]
Sorry, I will fix after some time.

**Comment by socketpair** [comment]
@ichernev month regexp should be sorted? AFAIK internal representation of compiled regexp throws order (in our case).

Proof:
```
$ echo qwerty | egrep -o 'we|wert'
wert
$ echo qwerty | egrep -o 'wert|we'
wert
```

**Comment by ichernev** [comment]
You're missing tests and a pipe character in the end of one of the lines.

File: src/locale/ru.js [hunk 1]
```diff
@@ -50,10 +50,15 @@ export default moment.defineLocale('ru', {
...
+
+    // полные названия с падежами, по три буквы, для некоторых, по 4 буквы, сокращения с точкой и без точки
+    monthsRegex: /^(январ[ья]|янв\.?|феврал[ья]|февр?\.?|марта?|мар\.?|апрел[ья]|апр\.?|ма[йя]|июн[ья]|июн\.?|июл[ья]|июл\.?|августа?|авг\.?|сентябр[ья]|сент?\.?|октябр[ья]|окт\.?|ноябр[ья]|нояб?\.?|декабр[ья]дек\.?)/i,
+    // копия предыдущего
+    monthsShortRegex: /^(январ[ья]|янв\.?|феврал[ья]|февр?\.?|марта?|мар\.?|апрел[ья]|апр\.?|ма[йя]|июн[ья]|июн\.?|июл[ья]|июл\.?|августа?|авг\.?|сентябр[ья]|сент?\.?|октябр[ья]|окт\.?|ноябр[ья]|нояб?\.?|декабр[ья]дек\.?)/i,
+    // полные названия с падежами
     monthsStrictRegex: /^(сентябр[яь]|октябр[яь]|декабр[яь]|феврал[яь]|январ[яь]|апрел[яь]|августа?|ноябр[яь]|марта?|июн[яь]|июл[яь]|ма[яй])/i,
-    monthsShortStrictRegex: /^(нояб\.|февр\.|сент\.|июль|янв\.|июн[яь]|мар[.т]|авг\.|апр\.|окт\.|дек\.|ма[яй])/i,
+    // Нет документации о том что здесь нужно писать. Оставляю так. с точками.
+    monthsShortStrictRegex: /^(янв\.|февр\.|мар[т.]|апр\.|ма[яй]|июн[ья.]|июл[ья.]|авг\.|сент\.|окт\.|нояб\.|дек\.)/i,
```
Comment (socketpair) [inline src/locale/ru.js]: Which forms must be written here? What mean "strict" ?

Comment (ichernev) [inline src/locale/ru.js]: Short strict means to only match short versions and not long ones.

**Comment by socketpair** [comment]
```

Running "jscs:all" (jscs) task
Warning: Config values to remove in 3.0:
The `esnext` option is enabled by default. Use --force to continue.
Aborted due to warnings.
```

**Comment by socketpair** [comment]
`>> 50 code style errors found!`

What the hell? not in files that I change! maybe I should rebase from another branch/tag ?

**Comment by socketpair** [comment]
@maggiepint I have fixed, but tests failed due to bugs not in my code (codestyle and build system).

**Comment by ichernev** [comment]
This will be merged in next release.

**Comment by socketpair** [comment]
YAY!

**Comment by ichernev** [comment]
Merged in 76f810837229d385cce8bc85e080f3c7ccb47931

**Pull request closed without merging**